flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -61,6 +72,13 @@
 
     @Nullable private final Predicate fieldPredicate;
 
+    /** The latest snapshot id seen at planning phase when manual compaction is triggered. */
+    @Nullable private final Long specifiedSnapshotId;
+
+    /** The manifest entries collected at planning phase when manual compaction is triggered. */
+    @Nullable
+    private transient Map<BinaryRowData, Map<Integer, List<DataFileMeta>>> specifiedManifestEntries;
```

Comment:
Can we just create a `SpecificDataFiles` class which implements `Serializable`.
And
- `Preconditions.checkNotNull(specifiedManifestEntries)` should in it.
- We dont need to have a `PartitionedDataManifestSerializer`.

Reply:
> Can we just create a `SpecificDataFiles` class which implements `Serializable`. And
> 
> * `Preconditions.checkNotNull(specifiedManifestEntries)` should in it.
> * We dont need to have a `PartitionedDataManifestSerializer`.

Do you mean creating a new serializable class like
```java
public class SpecificDataFiles implements Serializable {
    Map<BinaryRowData, Map<Integer, List<DataFileMeta>>>
}
```
Correct me if I'm wrong, but `BinaryRowData` and `DataFileMeta` are still not serializable, anyway we need to implement the ser/deser logic?